app/controllers/admin/auth_controller.rb

```diff
@@ -11,17 +11,36 @@ def sign_out
     end
 
     def callback
-      session[:login] = auth_hash.fetch("info").to_h
-      redirect_to admin_path
+      if authorised?
+        session[:login] = auth_hash.fetch("info").to_h
+        redirect_to admin_path
+      else
+        render "failure", status: :unauthorized
+      end
     end
 
     def failure
     end
 
     private
 
+    def authorised?
+      DfeSignin::Authorisation.new(
+        user_id: user_id,
+        organisation_id: organisation_id
+      ).has_role?(ENV["DFE_SIGN_IN_ROLE"])
```

Comment:
Can we pull the role out somewhere and verify that it's set at startup?

Reply:
I've tweaked it to do an `ENV.fetch`, that was it'll error on initialization if it's not set